### ISSUE TITLE
Missing Icon

### ISSUE THREAD

**Issue by rushsteve1** [flathub/com.slack.Slack#9]
The Window and Status  icons are missing. It shows the default fallback icons instead of the Slack icon.

Distro: Manjaro (though I have seen it on others)
DE: KDE Plasma 5.11

**Comment by nedrichards** [comment]
What version of flatpak are you using? (the output of `flatpak --version` in the terminal).

**Comment by rushsteve1** [comment]
Flatpak 0.10.3
Also I have just noticed that sometimes (seemingly arbitrarily) when I launch Slack the status icon is the correct Slack logo.

**Comment by nedrichards** [comment]
This sounds like a very similar issue to this one in Riot, which is also an Electron app. https://github.com/flathub/im.riot.Riot/issues/3

**Comment by Barthalion** [comment]
Yes, looks to be the same issue, as I haven't encountered it under GNOME on Arch.

**Comment by nedrichards** [comment]
This also may be fixed by 3.1 just released now (according to their upstream release notes) worth testing again.

**Comment by snj33v** [comment]
its not fixed

**Comment by Knusper** [comment]
Same issue here.

**Comment by snj33v** [comment]
this is fixed[using kde]

**Comment by Knusper** [comment]
Yes - the icon shows now in KDE, but it seems a lower resolution version than when installing the slack RPM from the website.

**Comment by liposo** [comment]
Same issue in Flatpak 1.0.2 on CentOS 7 using GNOME3 version 3.28.2

**Comment by krashekspres** [comment]
On xfce it's the same problem, both task switching and task manager icons are fallback.
Status panel icon is fine.
Flatpak 1.2.4
Slack 3.3.8
Xfce 4.12

**Comment by FakeShemp** [comment]
Is this still a problem with the latest update? I tried to solve it.

**Comment by krashekspres** [comment]
Nope, the only thing that fixes it on Xfce is a script which I run after Slack is running
```
#!/bin/bash
ICON=~/.local/share/flatpak/app/com.slack.Slack/current/active/export/share/icons/hicolor/512x512/apps/com.slack.Slack.png

WINDOWS=(`wmctrl -l | grep "Slack - " | cut -f1 -d' ' | xargs`)
for slack_window in ${WINDOWS[@]}; do
    # Use "xseticon", a compiled C binary, to change the icon of a running program
    /home/dario/bin/xseticon-0.1+bzr14/xseticon -id ${slack_window} $ICON

    # Use "xprop" to set the window state, so that alt+tab works again
    xprop -f _NET_WM_WINDOW_TYPE 32a -set _NET_WM_WINDOW_TYPE _NET_WM_WINDOW_TYPE_NORMAL -id ${slack_window}
done
```

**Comment by FakeShemp** [comment]
How about now?

**Comment by krashekspres** [comment]
Same.
I don't think you can fix it here, .rpm version has the same problem on xfce (tested it). Also "script" for fixing was proposed by Slack team. They are aware of some problems, not sure if they are fixing it though :))

**Comment by FakeShemp** [comment]
Yes, it seems that is the case. Thank you very much for testing it, though. Hopefully we will see a fix some time.

**Comment by FakeShemp** [comment]
Upstream mentioned a similar fix for Cinnamon in the 4.0 update. Perhaps it works without workarounds now?

**Comment by gone** [comment]
I'm seeing this with mate

**Comment by pzkpfwVI** [comment]
I have the same issue on debian 10.4

**Comment by Knusper** [comment]
Here I am also on debian 10.4, but in KDE, and the icon shows very nicely... also the high resolution version!

**Comment by Maltimore** [comment]
@Knusper for me it also shows the correct icon in the task bar, but when I switch windows with alt-tab, the icon for Slack (and other electron apps) is the generic "X" of the X Desktop. Is it the case for you as well? (I'm also on KDE)

**Comment by tored** [comment]
Missing icon for Ubuntu MATE 20.10

alt-tab: fallbacks Ubuntu MATE's icon for Appearance app.
taskbar: some default icon looking like a empty window

My guess, without knowing almost nothing about flatpaks and .desktop-files, is that Slack only ships with a 512x512 icon in this folder

`/var/lib/flatpak/app/com.slack.Slack/current/active/export/share/icons/hicolor`

Probably 512x512 is just too big in many cases.

If I compare that with the Bitwarden flatpak, that works for me, is has multiple icon sizes

`/var/lib/flatpak/app/com.bitwarden.desktop/current/active/export/share/icons/hicolor`

`128x128` `16x16` `256x256`  `32x32`  `48x48`  `512x512`  `64x64`

**Comment by jmayday** [comment]
```
➜  ~ flatpak --version
Flatpak 1.12.4
➜  ~ flatpak list  
Name                       Application ID                              Version    Branch       Installation
teams-for-linux            com.github.IsmaelMartinez.teams_for_linux   1.0.24     stable       system
Slack                      com.slack.Slack                             4.26.1     stable       system
Freedesktop Platform       org.freedesktop.Platform                    21.08.14   21.08        system
Mesa                       org.freedesktop.Platform.GL.default         21.3.8     21.08        system
Intel                      org.freedesktop.Platform.VAAPI.Intel                   21.08        system
openh264                   org.freedesktop.Platform.openh264           2.1.0      2.0          system
Breeze GTK theme           org.gtk.Gtk3theme.Breeze                    5.25.0     3.22         system
KDE Application Platform   org.kde.Platform                                       5.15-21.08   system
VLC                        org.videolan.VLC                            3.0.17.4   stable       system
➜  ~ 
```

![image](https://user-images.githubusercontent.com/10167533/174487401-5ecb1ae7-a7c3-44fd-8e68-fbc5677eaede.png)

**Comment by grongor** [comment]
I just installed it on Fedora; the icon in the taskbar is correct, but the one in the task switcher and "top left corner" is still the X.

**Comment by aardk** [comment]
Same issue (missing icon in taskbar) under Fedora38 KDE, solved by copying an svg icon for Slack to /var/lib/flatpak/exports/share/icons/hicolor/scalable/apps/

Note that the name of the icon should be com.slack.Slack.svg

**Comment by ZVNexus** [comment]
To fix this week need Slack to start reporting their app ID as "com.slack.Slack", or for KWin to workaround the issue similar to how they do it for taskbar.

**Comment by afiestas** [comment]
A workaround is to force the desktop file to com.slack.Slack with a window rule
![image](https://github.com/flathub/com.slack.Slack/assets/872727/51b1d145-231e-42f4-a3d8-dfe3d95b5781)

**Comment by ryaminal** [comment]
@afiestas, thank you for sharing that! definitely worked. had to restart the app(not certain if that's necessary) and it worked!

curious if anyone knows a CLI equivalent for this? with a kwin script or something? but be nice to share as this is probably a common workaround.

**Comment by Maltimore** [comment]
Are you already on KDE Plasma 6.0? Because it is fixed for me now, without me doing anything. I'm not 100% sure though that this came with the switch to Plasma 6.

**Comment by ZVNexus** [comment]
This is fixed if you are running the app under Wayland, still exists under XWayland though.
![image](https://github.com/user-attachments/assets/30aa9328-c411-4fc7-9b6b-15040278a924)
